artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPStandardMessage.java

```diff
@@ -235,10 +239,87 @@ public void reloadPersistence(ActiveMQBuffer record, CoreMessageObjectPools pool
 
       // Message state is now that the underlying buffer is loaded, but the contents not yet scanned
       resetMessageData();
+      recoverHeaderDataFromEncoding();
+
       modified = false;
       messageDataScanned = MessageDataScanningStatus.RELOAD_PERSISTENCE.code;
    }
 
+   private void recoverHeaderDataFromEncoding() {
+      final DecoderImpl decoder = TLSEncode.getDecoder();
+      decoder.setBuffer(data);
```

Comment:
I'm confused.. wasn't this done before?


Why this new code here?

Reply:
I mean.. I remember we would parse the header in out of order scenarios.. and we even had code.. I'm just wondering why you needed to add new code for that?

Reply:
The recover from persistence isn't reloading Header, or anything else for that matter before adding it to the Queue as the supplied test shows.  Messages read after restart all get added as if priority was set to default as I state in the issue if you happen to have read it.